libdrgn/lazy_parameter.c

```diff
@@ -45,3 +74,21 @@ struct drgn_error *drgn_lazy_parameter_get_type(struct drgn_lazy_parameter *lazy
 	return NULL;
 }
 
+struct drgn_error *drgn_lazy_parameter_get_object(struct drgn_lazy_parameter *lazy_parameter,
+						  struct drgn_object *ret) {
+	if (!drgn_lazy_parameter_is_evaluated(lazy_parameter)) {
+		struct drgn_object obj;
+		struct drgn_object_thunk *thunk_ptr = lazy_parameter->object_thunk;
+		struct drgn_object_thunk thunk = *thunk_ptr;
+		drgn_object_init(&obj, thunk.prog);
+		struct drgn_error *err = thunk.evaluate_fn(thunk_ptr, &obj);
+		if (err)
+			return err;
+		drgn_lazy_parameter_init_object(lazy_parameter, &obj);
+		thunk.free_fn(thunk_ptr);
+	}
+	drgn_object_init(ret, drgn_object_program(lazy_parameter->object));
```

Comment:
The convention in libdrgn is to assume that a passed object is already initialized. Rather than calling `drgn_object_init()` here, the caller should do it.